includes/api/class-wc-admin-rest-themes-controller.php

```diff
@@ -0,0 +1,190 @@
+<?php
+/**
+ * REST API Themes Controller
+ *
+ * Handles requests to /themes
+ *
+ * @package WooCommerce Admin/API
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Themes controller.
+ *
+ * @package WooCommerce Admin/API
+ * @extends WC_REST_Data_Controller
+ */
+class WC_Admin_REST_Themes_Controller extends WC_REST_Data_Controller {
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc-admin/v1';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'themes';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'upload_theme' ),
+					'permission_callback' => array( $this, 'upload_theme_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Check whether a given request has permission to edit upload plugins/themes.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function upload_theme_permissions_check( $request ) {
+		if ( ! current_user_can( 'upload_plugins' ) ) {
```

Comment:
Should this be `upload_themes` instead? https://wordpress.org/support/article/roles-and-capabilities/#upload_themes

Reply:
I felt I ran across `upload_plugins` when digging through the functions in core, so figure it might be some weird pseudo linked privilege.  But I can't seem to find it right now and `upload_themes` definitely seems better fitting.